src/storage/uncompressed_segment.cpp

```diff
@@ -118,11 +118,25 @@ UpdateInfo *UncompressedSegment::CreateUpdateInfo(ColumnData &column_data, Trans
 
 void UncompressedSegment::Fetch(ColumnScanState &state, idx_t vector_index, Vector &result) {
 	auto read_lock = lock.GetSharedLock();
-
 	InitializeScan(state);
 	FetchBaseData(state, vector_index, result);
 }
 
+//===--------------------------------------------------------------------===//
+// Filter
+//===--------------------------------------------------------------------===//
+bool UncompressedSegment::Select(Transaction &transaction, Vector &result, vector<TableFilter> &tableFilter,
+                                 SelectionVector &sel, idx_t &approved_tuple_count, ColumnScanState &state) {
+	auto read_lock = lock.GetSharedLock();
+	if (versions && versions[state.vector_index]) {
+		// TODO: We don't handle updates for filter selection during data fetching
+		return true;
```

Comment:
I think this propagation upwards if there are updates is unnecessarily complex: why not just run `UncompressedSegment::Scan` if there are any update to fetch the entire vector, and then run the filter expression on that vector here?